ffi/src/utils.rs

```diff
@@ -50,3 +39,16 @@ pub unsafe fn transform_credentials_handle<'a>(
         ))
     }
 }
+
+#[cfg(feature = "tsssp")]
+pub fn raw_wide_str_trim_nulls(raw_str: &mut Vec<u8>) {
+    let mut len = raw_str.len();
+    while len > 2 {
+        if raw_str[len - 2..] == [0, 0] {
+            raw_str.truncate(len - 2);
+            len = raw_str.len();
+        } else {
+            break;
+        }
+    }
```

Comment:
Maybe you could add a comment explaining that it works because when encoding an UTF-16 character using two code units, the 16-bit values are chosen from the UTF-16 surrogate range 0xD800–0xDFFF, and thus only `\0` is encoded by two consecutive null bytes.